src/V3Const.cpp

```diff
@@ -572,9 +572,9 @@ class ConstBitOpTreeVisitor final : public AstNVisitor {
                 incrOps(andp, __LINE__);
 
                 // Mark all bits checked by this comparison
-                const int maxBitIdx = std::min(ref.m_lsb + compNum.width(), ref.width());
-                for (int bitIdx = ref.m_lsb; bitIdx < maxBitIdx; ++bitIdx) {
+                for (int bitIdx = ref.m_lsb; bitIdx < ref.width(); ++bitIdx) {
                     const int maskIdx = bitIdx - ref.m_lsb;
+                    if (maskIdx >= maskNum.width()) break;
                     if (maskNum.bitIs0(maskIdx)) continue;
                     const bool polarity = compNum.bitIs1(maskIdx) != maskFlip;
```

Comment:
can `maskIdx` be out of range for `compNum.bitIs1` here? Do you need the same range check as you added below? Or you don't need it below either?